kvbc/include/merkle_processor/SyncMerkleProcessor.h

```diff
@@ -0,0 +1,43 @@
+// Concord
+//
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the
+// LICENSE file.
+
+#pragma once
+
+#include "BaseMerkleProcessor.h"
+
+namespace concord {
+namespace kvbc {
+namespace sparse_merkle {
+
+using concordUtils::bufferToHex;
+
+class SyncMerkleProcessor : public BaseMerkleProcessor {
+ public:
+  virtual ~SyncMerkleProcessor() = default;
+
+  virtual void Init(uint numVersionsStored) override;
+
+  virtual void BeginVersionUpdateBatch() override;
+  virtual void UpdateAccountTree(const address& addr, const std::string& key, const value& data) override;
+
+  virtual void CommitVersionUpdateBatch() override;
+
+  virtual std::vector<std::string> GetAccountMerkleRootPath(const address& addr) override;
+  virtual std::vector<std::string> GetAccountStorageKeyMerklePath(const address& addr, const std::string& key) override;
+  virtual std::string GetProof(const std::string& key) override;
```

Comment:
I personally find proofs being of type std::string to be very confusing. If you think of them as byte arrays, use the appropriate type (concord::Byte).

Reply:
In the response to GetProof for Ethereum we will have a JSON object as defined in [eth-getproof](https://docs.alchemy.com/reference/eth-getproof). What would be the benefit of replacing the std::string with an std::vector<concord::Byte>?

Reply:
The proof itself is defined by the hash bytes. The answer to the client request is these bytes encoded as hex strings. From a design perspective, types should match the held data. From a practical perspective - if we use raw byte arrays, when we need to reuse the proof data, we won't need to decode it from its hex string representation.